esx_service/vmdk_ops.py

```diff
@@ -129,6 +130,9 @@
 # For managing resource locks.
 lockManager = threadutils.LockManager()
 
+# regex for valid volume name
+VALID_VOLUME_NAME_REG = "[a-zA-Z0-9][a-zA-Z0-9_.-]*"
```

Comment:
docker local driver has this limitation on volume names during volume create. I think we should apply the same check.  We can discuss if you think differently. @msterin